sdk/scheduler/src/main/java/com/mesosphere/sdk/curator/CuratorStateStore.java

```diff
@@ -193,17 +192,27 @@ public void storeTasks(Collection<Protos.TaskInfo> tasks) throws StateStoreExcep
 
     @Override
     public void storeStatus(Protos.TaskStatus status) throws StateStoreException {
-        String taskName;
-        try {
-            taskName = CommonTaskUtils.toTaskName(status.getTaskId());
-        } catch (TaskException e) {
-            throw new StateStoreException(Reason.LOGIC_ERROR, String.format(
-                    "Failed to parse the Task Name from TaskStatus.task_id: '%s'", status), e);
+        Optional<String> optionalTaskName = Optional.empty();
+        for (Protos.TaskInfo taskInfo : fetchTasks()) {
+            if (taskInfo.getTaskId().getValue().equals(status.getTaskId().getValue())) {
+                if (optionalTaskName.isPresent()) {
+                    logger.error("Found duplicate taskIDs in Task {} and  Task {}",
+                            optionalTaskName.get(), taskInfo.getName());
+                    throw new StateStoreException(Reason.LOGIC_ERROR, String.format(
+                            "There are more than one tasks with TaskID: %s", status));
+                }
+                    optionalTaskName = Optional.of(taskInfo.getName());
+            }
         }
+        if (!optionalTaskName.isPresent()) {
+            throw new StateStoreException(Reason.NOT_FOUND, String.format(
+                    "Failed to find a task with TaskID: %s", status));
+        }
+        putStatus(optionalTaskName.get(), status);
+    }
 
-        // Validate that a TaskInfo with the exact same UUID is currently present. We intentionally
-        // ignore TaskStatuses whose TaskID doesn't (exactly) match the current TaskInfo: We will
-        // occasionally get these for stale tasks that have since been changed (with new UUIDs).
+    protected void putStatus(String taskName, Protos.TaskStatus status) throws StateStoreException {
+        //putStatus(taskName, status) is the only way to update status if there are duplicate taskIDs.
         Optional<Protos.TaskInfo> optionalTaskInfo;
         try {
             optionalTaskInfo = fetchTask(taskName);
```

Comment:
We're fetching the TaskInfo twice now.  In the loop above we already have the TaskInfo we want.